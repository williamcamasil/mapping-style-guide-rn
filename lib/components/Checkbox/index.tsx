import React, { useCallback } from 'react';
import {
  StyleProp,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native';

import { Images } from '../../assets';
import { useViewStyles } from '../../hooks/useStyles';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { getFieldErrorState, wrapField } from '../Form';
import Spacer from '../Spacer';
import Text, { TextPropTypes } from '../Text';
import {
  CheckboxVariantType,
  getBoxStyle,
  getTextColor,
} from './utils';

export type CheckBoxPropsType = {
  theme: AppThemeType;
  children: TextPropTypes['children'];
  testID?: string;
  onChecked?: (checked: boolean) => void;
  onBlur?: () => void;
  onFocus?: () => void;
  style?: StyleProp<ViewStyle>;
  error?: boolean;
  disabled?: boolean;
  checked?: boolean;
  variant?: CheckboxVariantType;
};

const Checkbox = ({
  testID,
  children,
  style,
  onChecked,
  onFocus,
  onBlur,
  error,
  disabled,
  checked,
  variant,
  theme,
}: CheckBoxPropsType): React.ReactElement => {

  const textColor = getTextColor(error, variant, disabled);

  const buttonStyles = useViewStyles(() => [
    {
      alignItems: 'center',
      flexDirection: 'row',
      flexShrink: 1,
    },
  ], []);

  const containerStyles = useViewStyles(() => [
    {
      flexDirection: 'row',
    },
    style,
  ], [style]);

  const boxStyles = getBoxStyle(theme, error, disabled, checked, variant);

  const handlePress = useCallback(() => {
    onFocus?.();
    onChecked?.(!checked);
    onBlur?.();
  }, [checked, onBlur, onChecked, onFocus]);

  return (
    <View style={containerStyles}>
      <TouchableOpacity
        testID={testID}
        onPress={handlePress}
        disabled={disabled}
        activeOpacity={theme.opacities.intense}
        style={buttonStyles}
      >
        <Images.CheckboxImg {...boxStyles} />
        <Spacer size={theme.spacings.sXXS} />
        <Text
          color={textColor}
          variant="body"
        >
          {children}
        </Text>
      </TouchableOpacity>
    </View>
  );
};

Checkbox.defaultProps = {
  variant: 'default' as CheckboxVariantType,
};

Checkbox.Field = withTheme(wrapField<CheckBoxPropsType, boolean | undefined>((props, fieldProps) => {
  const {
    invalid, touched, value, onChange, onBlur,
    onFocus,
  } = fieldProps;

  const { showError } = getFieldErrorState(
    invalid,
    touched,
  );

  return (
    <Checkbox
      {...props}
      checked={value || false}
      onChecked={onChange}
      onBlur={onBlur}
      onFocus={onFocus}
      error={showError}
    />
  );
}));

export default withTheme(Checkbox);
