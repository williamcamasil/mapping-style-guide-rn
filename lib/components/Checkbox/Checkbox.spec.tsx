import React from 'react';

import { fireEvent, render } from '@testing-library/react-native';
import { createForm } from 'final-form';

import Checkbox from '.';
import Form from '../Form';

describe('checkbox snapshot', () => {
  it('Should render checkbox', () => {
    const tree = render(
      <Checkbox onChecked={() => null}>
        Clique aqui
      </Checkbox>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render checkbox with secondary background', () => {
    const tree = render(
      <Checkbox onChecked={() => null} variant="secondary">
        Clique aqui
      </Checkbox>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render checkbox disabled', () => {
    const tree = render(
      <Checkbox disabled onChecked={() => null}>
        Clique aqui
      </Checkbox>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render checkbox disabled with secondary background', () => {
    const tree = render(
      <Checkbox disabled onChecked={() => null} variant="secondary">
        Clique aqui
      </Checkbox>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render checkbox with error', () => {
    const tree = render(
      <Checkbox error onChecked={() => null}>
        Clique aqui
      </Checkbox>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render checkbox with error and secondary background', () => {
    const tree = render(
      <Checkbox error onChecked={() => null} variant="secondary">
        Clique aqui
      </Checkbox>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render checkbox checked', () => {
    const tree = render(
      <Checkbox checked onChecked={() => null}>
        Clique aqui
      </Checkbox>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render checkbox checked with secondary background', () => {
    const tree = render(
      <Checkbox checked onChecked={() => null} variant="secondary">
        Clique aqui
      </Checkbox>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render checkbox checked and with error', () => {
    const tree = render(
      <Checkbox checked error onChecked={() => null}>
        Clique aqui
      </Checkbox>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render checkbox using form', () => {
    const result = render(
      <Form
        onSubmit={jest.fn()}
      >
        {() => (
          <Checkbox.Field
            name="checkboxName"
          >
            Clique aqui
          </Checkbox.Field>
        )}
      </Form>,
    );

    const tree = result.toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Checkbox events', () => {
  it('Should test checkbox action', () => {
    const mockFunction = jest.fn();
    const tree = render(
      <Checkbox
        onChecked={mockFunction}
        testID="touchable-checkbox"
      >
        Clique aqui
      </Checkbox>,
    );

    const primaryLinkInstance = tree.getByTestId('touchable-checkbox');

    fireEvent.press(primaryLinkInstance);

    expect(mockFunction).toHaveBeenCalled();
  });

  it('Should test checkbox action checked', () => {
    const mockFunction = jest.fn();
    const tree = render(
      <Checkbox
        onChecked={mockFunction}
        checked
        testID="touchable-checkbox"
      >
        Clique aqui
      </Checkbox>,
    );

    const primaryLinkInstance = tree.getByTestId('touchable-checkbox');

    fireEvent.press(primaryLinkInstance);

    expect(mockFunction).toHaveBeenCalled();
  });

  it('Should test checkbox action disable', () => {
    const mockFunction = jest.fn();
    const tree = render(
      <Checkbox
        onChecked={mockFunction}
        disabled
        testID="touchable-checkbox"
      >
        Clique aqui
      </Checkbox>,
    );

    const primaryLinkInstance = tree.getByTestId('touchable-checkbox');

    fireEvent.press(primaryLinkInstance);

    expect(mockFunction).not.toHaveBeenCalled();
  });

  it('Should test checkbox returning checked value', () => {
    const mockFunction = jest.fn();
    const tree = render(
      <Checkbox
        onChecked={mockFunction}
        checked={false}
        testID="touchable-checkbox"
      >
        Clique aqui
      </Checkbox>,
    );

    const primaryLinkInstance = tree.getByTestId('touchable-checkbox');

    fireEvent.press(primaryLinkInstance);

    expect(mockFunction).toHaveBeenCalledWith(true);
  });

  it('Should test checkbox returning unchecked value', () => {
    const mockFunction = jest.fn();
    const tree = render(
      <Checkbox
        onChecked={mockFunction}
        checked
        testID="touchable-checkbox"
      >
        Clique aqui
      </Checkbox>,
    );

    const primaryLinkInstance = tree.getByTestId('touchable-checkbox');

    fireEvent.press(primaryLinkInstance);

    expect(mockFunction).toHaveBeenCalledWith(false);
  });
});

describe('Checkbox.Field', () => {
  it('Should test form invalid', () => {
    const form = createForm({
      onSubmit: jest.fn(),
    });

    render(
      <Form
        onSubmit={jest.fn()}
        form={form}
      >
        {() => (
          <Checkbox.Field
            name="checkboxName"
            required
            validate={checked => (checked ? undefined : 'Erro')}
            testID="checkbox-field"
          >
            Clique aqui
          </Checkbox.Field>
        )}
      </Form>,
    );

    expect(form.getState().invalid).toBeTruthy();
  });

  it('Should test form valid', () => {
    const form = createForm({
      onSubmit: jest.fn(),
    });

    const result = render(
      <Form
        onSubmit={jest.fn()}
        form={form}
      >
        {() => (
          <Checkbox.Field
            name="checkboxName"
            required
            validate={checked => (checked ? undefined : 'Erro')}
            testID="checkbox-field"
          >
            Clique aqui
          </Checkbox.Field>
        )}
      </Form>,
    );

    const checkbox = result.getByTestId('checkbox-field');

    fireEvent.press(checkbox);

    expect(form.getState().valid).toBeTruthy();
  });
});
