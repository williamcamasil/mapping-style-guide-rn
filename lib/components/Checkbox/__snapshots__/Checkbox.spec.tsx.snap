// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox snapshot Should render checkbox 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4294440952,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4293482516,
              "type": 0,
            }
          }
          strokeWidth="1"
          width="22"
          x="1"
          y="1"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#3E4149",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`checkbox snapshot Should render checkbox checked 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4293482516,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4293980402,
              "type": 0,
            }
          }
          strokeWidth="2"
          width="22"
          x="1"
          y="1"
        />
        <RNSVGPath
          d="m16.01 10-5.414 4.67L8 12.43"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#3E4149",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`checkbox snapshot Should render checkbox checked and with error 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4293212464,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4293980402,
              "type": 0,
            }
          }
          strokeWidth="2"
          width="22"
          x="1"
          y="1"
        />
        <RNSVGPath
          d="m16.01 10-5.414 4.67L8 12.43"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#E53930",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`checkbox snapshot Should render checkbox checked with secondary background 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4293482516,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4280570793,
              "type": 0,
            }
          }
          strokeWidth="2"
          width="22"
          x="1"
          y="1"
        />
        <RNSVGPath
          d="m16.01 10-5.414 4.67L8 12.43"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#C6D3EA",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`checkbox snapshot Should render checkbox disabled 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 0,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4293980402,
              "type": 0,
            }
          }
          strokeWidth="1"
          width="22"
          x="1"
          y="1"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#E1E2E4",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`checkbox snapshot Should render checkbox disabled with secondary background 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 0,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4283201216,
              "type": 0,
            }
          }
          strokeWidth="1"
          width="22"
          x="1"
          y="1"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#4C76C0",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`checkbox snapshot Should render checkbox using form 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4294440952,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4293482516,
              "type": 0,
            }
          }
          strokeWidth="1"
          width="22"
          x="1"
          y="1"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#3E4149",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`checkbox snapshot Should render checkbox with error 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4294440952,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4293212464,
              "type": 0,
            }
          }
          strokeWidth="1"
          width="22"
          x="1"
          y="1"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#E53930",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`checkbox snapshot Should render checkbox with error and secondary background 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4279189913,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4294938248,
              "type": 0,
            }
          }
          strokeWidth="1"
          width="22"
          x="1"
          y="1"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#FF8E88",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;

exports[`checkbox snapshot Should render checkbox with secondary background 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGRect
          fill={
            {
              "payload": 4279189913,
              "type": 0,
            }
          }
          height="22"
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          rx="5"
          stroke={
            {
              "payload": 4293482516,
              "type": 0,
            }
          }
          strokeWidth="1"
          width="22"
          x="1"
          y="1"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        {
          "height": 12,
          "width": 12,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#C6D3EA",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      Clique aqui
    </Text>
  </View>
</View>
`;
